### ISSUE TITLE
Sorting characters by popularity is strange and contains houses

### ISSUE THREAD

**Issue by mammuth** [Rostlab/JS16_ProjectF#224]
Is anyone of us responsible for this, or has been something changed on the API? 
@jorjo1 @yashha @mina-zaki @sacdallago 
<img width="1291" alt="screen shot 2016-04-01 at 16 26 49" src="https://cloud.githubusercontent.com/assets/3121306/14209869/97bdc78c-f826-11e5-8fb9-c827f0eef503.png">

**Comment by mina-kleid** [comment]
Thats weird, i will take a look at it

**Comment by sacdallago** [comment]
https://github.com/Rostlab/JS16_ProjectA/issues/126

**Comment by gyachdav** [comment]
I believe @Adiolis changed loaded a new pageRank today. this may have messed up the sorting (although it shouldnt have since the only difference was to normalize the scores not resort). 

@Adiolis can you comment?

**Comment by sacdallago** [comment]
Not it's not just that, it's also the issue on houses getting scraped as characters although they aren't.

I have two takes on this:
1. I updated the data, but the old pagerank gave these characters and the houses a now higher value
2. Since these characters don't have a pagerank they appear either at the end or the beginning of the sorted list and you ( **F** ) need to sort that out.

Look into it and report pls

**Comment by gyachdav** [comment]
do you know why the pagerank changed?

I actually think that A need to handle the case of characters with missing pagerank just assign some low default. if there is an image bump the pagerank. i'll commented on the issue https://github.com/Rostlab/JS16_ProjectA/issues/103

This will all be resolved once we have the centrality measure feature in place. possibly only in JS6-2 :worried:

**Comment by sacdallago** [comment]
The pagerank changed because there were some chars set as super-important although they counted relatively little.

I think that not assigning a pagerank is good a statement already! As I said, this is a bunch of data that will appear either in front or back of the sorted response from A, but F can handle this with a filter and put it in the back always, which is reasonable, considering that if there's no page rank it's either miss-scraped or totally unimportant.

Probably, yes. Not gonna be around to see that though :bikini: :D

**Comment by gyachdav** [comment]
@mina-zaki please handle asap then.

The houses showing up in the list will be corrected by @boriside from A

**Comment by mina-kleid** [comment]
@gyachdav if they are corrected the character list then this bug will disapear, because i sort and filter on the character list

**Comment by gyachdav** [comment]
Do you put the characters with no ranking at the bottom of the list? Do you know why we have a bunch of unknown characters now at the first losing page event high we sort by rank?

**Comment by boriside** [comment]
I cleaned the illegal character names, but now the whole issue #126 about the task is missing.

**Comment by mina-kleid** [comment]
@sacdallago @boriside 
Now am confused with the pageRank. there are values as  0.01 and values as 64. shall i use it? and which is the highest in the rank ? 1 or 64 or 0.01 ?

**Comment by sacdallago** [comment]
I guess the update on the collection yesterday only updated pageranks without deleting the new ones, I might have to rescrape the whole thing.

@adiolis can you confirm?

**Comment by ghost** [comment]
That's why i asked if the new pagerank file contains as much data as the old one 😉. Basically you could run a refill before.

**Comment by sacdallago** [comment]
I'm running: 

``` bash
npm run refill --collection=characters && npm run updatePageRanks --update=characters --file=data/normalizedPageRanks.json
```

this might take some time (5 min) Then @mina-zaki please check again.

**Comment by sacdallago** [comment]
It's still running... It seems to be stuck.. I'll update you when this is really done @mina-zaki sorry

**Comment by sacdallago** [comment]
@Adiolis I constantly get stuck on this:

``` bash
Still 9 characters to fetch.
Fetched Zei
Still 8 characters to fetch.
Fetched Zekko
Still 7 characters to fetch.
Fetched Zharaq zo Loraq
Still 6 characters to fetch.
Fetched Zachery Frey
Still 5 characters to fetch.
Fetched Zollo
Still 4 characters to fetch.
Fetched Zia Frey
Still 3 characters to fetch.
Fetched Zhoe Blanetree
Still 2 characters to fetch.
```

at the `npm run refill --collection=characters` stage. What should I do? Delete the entire collection and scrape from zero?

**Comment by ghost** [comment]
Isn´t @boriside skipping 2 characters, because they are houses? This should be fixed in the getAllNames function and not in getAll...

I am trying it out, right now.

**Comment by ghost** [comment]
@sacdallago Yeah, this skipping was the problem.
I fixed it now. Please pull and rerun ;)

**Comment by sacdallago** [comment]
Ok. Done and it terminated. Please @mina-zaki have a look and report back.

**Comment by gyachdav** [comment]
@sacdallago is the rank still according to the list I sent or now using the normalized value?

**Comment by sacdallago** [comment]
It's according to https://github.com/Rostlab/JS16_ProjectA/issues/103#issuecomment-203649730

**Comment by gyachdav** [comment]
okay @sacdallago @mina-zaki @kordianbruck @Adiolis I need all of you to focus for a moment and figure this out. The sorting of the characters are still messed up. 

If I look at  the characters that show up in the first page I get for instance Dalla (Dragonstone). when searching the API  https://api.got.show/api/characters/Dalla%20(Dragonstone) I get  No character with that data existing!

Moving on, if I search for Visenya Targaryen (also currently on the first listing page) https://api.got.show/api/characters/Visenya%20Targaryen. 
her rank is still outside the[0-1] range (it's actually 84). 

Same for Qyburn. his page rank is 161. 

So in summary it seems like some of the characters are either missing on the api (or im not formatting special chars correctly in my API call) AND much more importantly **the page rank data is corrupt and the refill needs to be redone.**

**Comment by mina-kleid** [comment]
As far as i see now all page ranks are outside [0-1], and some of them have no page rank.
From our side the higher the page rank the more popular it is, and people without page rank go to the end of the list. 
@sacdallago If this enough for the requirement please close this issue

**Comment by sacdallago** [comment]
Yes. You got it right @mina-zaki
